### PR TITLE
Add Dockerfiles for DockerHub autobuilt containers

### PR DESCRIPTION
**Pull request by tjesser-ucdavis-edu**
Once set up, the DockerHub autobuilt containers will be on the [Geodynamics DockerHub](https://hub.docker.com/u/geodynamics/) page

Credit to @gassmoeller for the Dockerfiles

### CODE REVIEW

File: docker/rayleigh-buildenv-bionic/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  apt install --yes \
+    g++ \
+    gfortran \
+    git \
+    libblas-dev \
+    libfftw3-dev \
+    liblapack-dev \
+    libopenmpi-dev \
+    make \
+    nano
```
Comment (gassmoeller) [inline docker/rayleigh-buildenv-bionic/Dockerfile]: Coud you add `wget`? We will need it later on in the testing.

File: docker/rayleigh/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+USER rayleigh_user
+
+WORKDIR /home/rayleigh_user
+
+RUN git clone 'https://github.com/geodynamics/Rayleigh.git'
+
+WORKDIR /home/rayleigh_user/Rayleigh
+
+RUN git checkout devel \
```
Comment (gassmoeller) [inline docker/rayleigh/Dockerfile]: this first step is no longer necessary, `master` is now the development branch, and `devel` will be removed in the future. Just the `configure` step should work fine.

**Comment by gassmoeller** [comment]
Looks good, thanks :+1:. For the future: Could you leave me a little comment when you have addressed comments? Otherwise I am never sure, if you are still working on some parts of my comments, or if everything is already addressed.

**Comment by tjesser-ucdavis-edu** [comment]
Ah, sorry. I'll do that from now on.

**Pull request merged**